### ISSUE TITLE
Add Flutter template

### ISSUE THREAD

**Issue by ndonkoHenri** [project-idx/templates#12]
Can we please have the flutter template files here too?

**Comment by romannurik** [comment]
Hi there! We're working on it! Unfortunately the Flutter template is the only one that's a bit more involved than just a simple template. We're working on bringing the Flutter environment 100% onto our templating system, which is taking some time.

**Comment by mg3994** [comment]
@romannurik  [Flutter](https://github.com/project-idx/community-templates/tree/main/flutter-create), please check this link , I found most of part useless,

**Comment by romannurik** [comment]
@mg3994 We have a number of alternative/secondary approaches to spinning up Flutter workspaces that are unofficial/experimental (that's one of them) ... but the default "new Flutter app" experience on IDX isn't currently replicable purely via custom templates unfortunately.

**Comment by rodydavis** [comment]
The flutter template I created can get pretty close to the environment with the default flutter template but still not 100%.

**Comment by romannurik** [comment]
yeah, my understanding also is the default flutter template is hyper-optimized for startup performance, using mechanisms unavailable to other templates at the moment.

**Comment by rodydavis** [comment]
Yep but if speed is not the concern then the template I shared will still have flutter fully working with the previews on Android and web 👍🏼

**Comment by mikeborodin** [comment]
I saw that nixpkg `flutter` package is not up to date, interesting how this is solved in idx, probably some custom nix flake?

**Comment by rodydavis** [comment]
I only use the flutter nix to bootstrap the files.

**Comment by rodydavis** [comment]
Also the clarify the flutter template I create is just as fast running and is still super quick opening.

It adds a lot of the flutter create commands to the template instead of caching the flutter counter app only.
